oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java

```diff
@@ -254,7 +254,13 @@ private AsyncRefreshResult getOrCreateRefreshTask() {
               new Callable<OAuthValue>() {
                 @Override
                 public OAuthValue call() throws Exception {
-                  return OAuthValue.create(refreshAccessToken(), getAdditionalHeaders());
+                  OAuthValue result = OAuthValue.create(refreshAccessToken(), getAdditionalHeaders());
+
+                  synchronized (lock) {
```

Comment:
Is this lock necessary? Only one thread should be doing work on this future right?

Reply:
The lock was taken in the `finishRefreshAsync()` method when modifying the `value` variable. I wasn't sure why, if it was needed to protect the `value` variable or something else. But you are probably right, the lock is not necessary here.